pandas/core/tools/datetimes.py

```diff
@@ -71,26 +120,19 @@ def _convert_and_box_cache(arg, cache_array, box, errors, name=None):
         Cache of converted, unique dates
     box : boolean
         True boxes result as an Index-like, False returns an ndarray
-    errors : string
-        'ignore' plus box=True will convert result to Index
     name : string, default None
         Name for a DatetimeIndex
 
     Returns
     -------
     result : datetime of converted dates
-        Returns:
-
         - Index-like if box=True
         - ndarray if box=False
     """
-    from pandas import Series, DatetimeIndex, Index
+    from pandas import Series
     result = Series(arg).map(cache_array)
     if box:
-        if errors == 'ignore':
-            return Index(result, name=name)
-        else:
-            return DatetimeIndex(result, name=name)
+        return _box_as_indexlike(result, tz=None, name=name)
```

Comment:
Do we have a test where `cache=True` and `utc=True`? I would think `tz` needs to be passed here instead of `None` if that were the case.

Reply:
@anmyachev do we have a test like this? 

you are passing tz in the call below

Reply:
> @anmyachev do we have a test like this? 

@jreback yes. For example, `test_to_datetime_tz_pytz` (`pandas\tests\indexes\datetimes\test_tools.py::519`)

```python
    @pytest.mark.parametrize('cache', [True, False])
    def test_to_datetime_tz_pytz(self, cache):
        # see gh-8260
        us_eastern = pytz.timezone('US/Eastern')
        arr = np.array([us_eastern.localize(datetime(year=2000, month=1, day=1,
                                                     hour=3, minute=0)),
                        us_eastern.localize(datetime(year=2000, month=6, day=1,
                                                     hour=3, minute=0))],
                       dtype=object)
        result = pd.to_datetime(arr, utc=True, cache=cache)
        expected = DatetimeIndex(['2000-01-01 08:00:00+00:00',
                                  '2000-06-01 07:00:00+00:00'],
                                 dtype='datetime64[ns, UTC]', freq=None)
        tm.assert_index_equal(result, expected)
```

Reply:
I c, we call the passed parameter tz, which is sligthly confusing.